File: 200-number-of-islands/number-of-islands.java
class Solution {
    void helper(char[][] grid, int i , int j){
        if(i < 0 || j < 0 || i >= grid.length || j >= grid[0].length || grid[i][j] != '1'){
            return;
        }
        grid[i][j] = '2';
        helper(grid, i + 1, j);
        helper(grid, i - 1, j);
        helper(grid, i, j + 1);
        helper(grid, i, j - 1);

    }
    public int numIslands(char[][] grid) {
        int i = 0;
        int j = 0;
        int result = 0;
        int m = grid.length;
        int n = grid[0].length;
        while(i < m){
            j = 0;
            while(j < n){
                if(grid[i][j] == '1'){
                    helper(grid, i , j);
                    result++;
                }
                j++;
            }
            i++;
        }
        return result;

    }
}

File: 200-number-of-islands/number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        result = 0
        curr = 0
        def helper(grid, i, j):
            if i < 0 or j < 0 or i >= len(grid) or j >= len(grid[0]) or grid[i][j] != '1':
                return
            grid[i][j] = '*'
            helper(grid, i + 1, j)
            helper(grid, i - 1, j)
            helper(grid, i, j + 1)
            helper(grid, i, j - 1)    
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == "1":
                    result = result + 1
                    helper(grid, i, j)
        return result
        
        
        